File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/Java/Chapter5/src/chapter5/StackLinkedListChar.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package chapter5;

public class StackLinkedListChar {

    private NodeChar top;

    public StackLinkedListChar() {
        top = null;
    }

    public boolean isEmpty() {
        return top == null;
    }

    public void push(char newItem) {
        top = new NodeChar(newItem, top);
    }

    public char pop() {
        if (!isEmpty()) {
            NodeChar temp = top;
            top = top.getNext();
            return temp.getItem();
        } else {
            return 'j';
        }
    }

    public void popAll() {
        top = null;
    }

    public char peek() {
        if (!isEmpty()) {
            return top.getItem();
        } else {
            return 'j';
        }
    }
}

File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/C/StackLinkedListChar.cpp
#include <stdio.h>
#include <NodeChar.h>
#define true 1
#define false 0
typedef int boolean;

struct NodeChar *top = NULL;
boolean isEmpty(){
  if(top == NULL)
    return true;
  else return false;
}
void push(char newItem){
  top = insertNode(newItem,top);
}
char pop(){
  if(isEmpty() == false){
    struct NodeChar *temp = top;
    top = top->next;
    return temp->item;
  }else return '1';
}
void popAll(){
  top = NULL;
}
char peek(){
  if(isEmpty() == false){
    return top->item;
  }return -1;
}